### PR TITLE
Blog-app: Integration test and n+1 problem fixing

### PR DESCRIPTION
**Pull request by GedeonTS**
## Blog-app:Integration test and n+1 problem fixing

In this pull request I:
- Solved N+1 problem
- Formed a feature folder that holds the integration test of the app
- The integration test includes:
  - The users index page
  - The user show page
  - The posts index page
  - the post show page

### CODE REVIEW

File: spec/views/posts_spec.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    visit user_post_path(@user, @user.posts.first)
+    expect(page).to have_content('Tom')
+  end
+
+  it 'Should content See all posts button' do
+    visit user_path(@user.id)
+    expect(page).to have_content('Title')
+    expect(page).to have_content('See all posts')
+  end
+end
```
Comment (AnselemOdims) [inline spec/views/posts_spec.rb:47]: - [ ] Please kindly include tests that assert the following 
  - You can see the first comments on a post.
  - You can see how many likes a post has.
  - You can see a section for pagination if there are more posts than fit on the view
  - Redirects to the post's show page when you click on a post.

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @GedeonTS, 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: spec/views/posts_spec.rb (hunk 1 above)
Comment (GedeonTS) [reply spec/views/posts_spec.rb:47]: Thank you very much @AnselemOdims, let me do it right away!

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 @GedeonTS, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- You have done a great job working on the requested changes✔️✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**